platform_plugin_turnitin/turnitin.py

```diff
@@ -0,0 +1,427 @@
+    def upload_turnitin_submission_file(self, data, suffix=""):
+        """
+        Handles the upload of the user's file to Turnitin.
+
+        Args:
+            data (WebRequest): Web request containing the file to be uploaded.
+            suffix (str, optional): Additional suffix for the request. Defaults to ''.
+
+        Returns:
+            Response: The response after uploading the file to Turnitin.
+        """
+        turnitin_submission = self.create_turnitin_submission_object()
+        if turnitin_submission.status_code == 201:
+            turnitin_submission_id = turnitin_submission.json()["id"]
+            current_user_id = self.get_user_data()["user_id"]
+            current_user = User.objects.get(id=current_user_id)
+            submission = TurnitinSubmission(
+                user=current_user, turnitin_submission_id=turnitin_submission_id
+            )
+            submission.save()
+            myfile = data.params["myfile"].file
+            # turnitin_submission_id='0a966646-83f9-4ce6-aa47-71e07baf4e30'
+            response = put_upload_submission_file_content(
+                turnitin_submission_id, myfile
+            )
+            return Response(
+                json.dumps(response.json()),
+                content_type="application/json",
+                charset="UTF-8",
+            )
+        return Response(
+            json.dumps(turnitin_submission.json()),
+            content_type="application/json",
+            charset="UTF-8",
+        )
+
+    @XBlock.json_handler
+    def get_submission_status(self, data, suffix=""):
+        """
+        Retrieves the status of the latest Turnitin submission for the user.
+
+        Args:
+            data (dict): Input data for the request.
+            suffix (str, optional): Additional suffix for the request. Defaults to ''.
+
+        Returns:
+            dict: Information related to the user's latest Turnitin submission.
+        """
+        current_user_id = self.get_user_data()["user_id"]
+        current_user = User.objects.get(id=current_user_id)
+        try:
+            last_submission = TurnitinSubmission.objects.filter(
+                user=current_user
+            ).latest("created_at")
+        except TurnitinSubmission.DoesNotExist:
+            return {"success": False}
+        # last_submission = 'de6784c5-471f-4220-aff1-16b6b44dffcf'
+        response = get_submission_info(last_submission.turnitin_submission_id)
+        return response.json()
+
+    @XBlock.json_handler
+    def generate_similarity_report(self, data, suffix=""):
+        """
+        Initiates the generation of a similarity report for the user's latest Turnitin submission.
+
+        Args:
+            data (dict): Input data for the request.
+            suffix (str, optional): Additional suffix for the request. Defaults to ''.
+
+        Returns:
+            dict: The status of the similarity report generation process.
+        """
+
+        payload = {
```

Comment:
what do these settings mean? should we have a plugin setting called TURNITIN_SIMILARITY_REPORT_SETTINGS for this?

Reply:
@mariajgrimaldi Yes... we should define each parameter from the optional settings that Turnitin offers to obtain the result we expect from them.

Reply:
@mariajgrimaldi The only issue is that there are nested variables throughout the request structure. So in order to create the settings variable, we have to check the docs to create many variables as need and put them  throughout the request structure. We need more time and owner's review to do this